File: sdk-python/examples/forms/update.py
"""Run with: python examples/forms/update.py

Only the fields you set on ``UpdateBody`` are changed; everything else is
left untouched.
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.forms import UpdateBody


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        updated = client.forms.update(
            "frm_replace_with_real_id",
            UpdateBody(
                name="Updated Registration",
                status="active",
                submit_button_text="Sign up",
            ),
        )
        print(f"updated {updated.id} -> {updated.name} ({updated.status})")


if __name__ == "__main__":
    main()

File: sdk-node/examples/forms/update.ts
/**
 * Update a form's settings (publish it and customize the submit button).
 *
 * Run:
 *   npx tsx examples/forms/update.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const form = await client.forms.update('frm_replace_with_real_id', {
  name: 'Updated Registration',
  status: 'active',
  submitButtonText: 'Sign up',
})

console.log(`updated ${form.id} - ${form.name} (${form.status})`)
